packages/cloud_firestore/ios/Classes/FirestoreCodecs.mm

```diff
@@ -0,0 +1,441 @@
+               strcmp(type, @encode(unsigned char)) == 0) {
+      SInt32 n = number.intValue;
+      [self writeByte:FirestoreFieldInt32];
+      [_data appendBytes:(UInt8*)&n length:4];
+    } else if (strcmp(type, @encode(signed long)) == 0 ||
+               strcmp(type, @encode(unsigned int)) == 0) {
+      SInt64 n = number.longValue;
+      [self writeByte:FirestoreFieldInt64];
+      [_data appendBytes:(UInt8*)&n length:8];
+    } else if (strcmp(type, @encode(double)) == 0 || strcmp(type, @encode(float)) == 0) {
+      Float64 f = number.doubleValue;
+      [self writeByte:FirestoreFieldFloat64];
+      [self writeAlignment:8];
+      [_data appendBytes:(UInt8*)&f length:8];
+    } else if (strcmp(type, @encode(unsigned long)) == 0 ||
+               strcmp(type, @encode(signed long long)) == 0 ||
+               strcmp(type, @encode(unsigned long long)) == 0) {
+      NSString* hex = [NSString stringWithFormat:@"%llx", number.unsignedLongLongValue];
+      [self writeByte:FirestoreFieldIntHex];
+      [self writeUTF8:hex];
+    } else {
+      NSLog(@"Unsupported value: %@ of type %s", value, type);
+      NSAssert(NO, @"Unsupported value for standard codec");
+    }
+  } else if ([value isKindOfClass:[NSString class]]) {
+    NSString* string = value;
+    [self writeByte:FirestoreFieldString];
+    [self writeUTF8:string];
+  } else if ([value isKindOfClass:[FlutterStandardBigInteger class]]) {
+    FlutterStandardBigInteger* bigInt = value;
+    [self writeByte:FirestoreFieldIntHex];
+    [self writeUTF8:bigInt.hex];
+  } else if ([value isKindOfClass:[FlutterStandardTypedData class]]) {
+    FlutterStandardTypedData* typedData = value;
+    [self writeByte:FirestoreFieldForDataType(typedData.type)];
+    [self writeSize:typedData.elementCount];
+    [self writeAlignment:typedData.elementSize];
+    [_data appendData:typedData.data];
+  } else if ([value isKindOfClass:[NSArray class]]) {
+    NSArray* array = value;
+    [self writeByte:FirestoreFieldList];
+    [self writeSize:array.count];
+    for (id object in array) {
+      [self writeValue:object];
+    }
+  } else if ([value isKindOfClass:[NSDictionary class]]) {
+    NSDictionary* dict = value;
+    [self writeByte:FirestoreFieldMap];
+    [self writeSize:dict.count];
+    for (id key in dict) {
+      [self writeValue:key];
+      [self writeValue:[dict objectForKey:key]];
+    }
+  } else if ([value isKindOfClass:[NSDate class]]) {
+    NSDate* date = value;
+    NSNumber seconds = [value timeIntervalSince1970] * 1000;
+    SInt64 n = [seconds longValue];
+    [self writeByte:FirestoreFieldDateTime];
+    [_data appendBytes:(UInt8*)&n length:8];
+  } else if ([value isKindOfClass:[FIRFieldValue]]) {
+    if ([value isKindOfClass:[FSTDeleteFieldValue]]) {
+      SInt32 n = 0;
+      [self writeByte:FirestoreFieldFieldValue];
+      [_data appendBytes:(UInt8*)&n length:4];
+    } else if ([value isKindOfClass:[FSTServerTimestampFieldValue]]) {
```

Comment:
`FSTDeleteFieldValue` and `FSTServerTimestampFieldValue` are pulled direct from [here](https://github.com/firebase/firebase-ios-sdk/blob/bde743ed25166a0b320ae157bfb1d68064f531c9/Firestore/Source/API/FIRFieldValue%2BInternal.h)

Reply:
@Skylled 
I've just refined iOS part, and give you a pull request on your `patch-3` branch 😸